django/db/backends/mysql/schema.py

```diff
@@ -29,21 +29,25 @@ def quote_value(self, value):
         import MySQLdb.converters
         return MySQLdb.escape(value, MySQLdb.converters.conversions)
 
-    def skip_default(self, field):
+    def _limited_data_types(self, field):
         """
-        MySQL doesn't accept default values for some data types and implicitly
-        treats these columns as nullable.
+        MySQL have a limited functionality for some data types, i.e.:
+        - doesn't accept default values and implicitly treats these columns
+          as nullable,
+        - doesn't support a database index.
         """
         db_type = field.db_type(self.connection)
         return (
             db_type is not None and
-            db_type.lower() in {
-                'tinyblob', 'blob', 'mediumblob', 'longblob',
-                'tinytext', 'text', 'mediumtext', 'longtext',
-                'json',
-            }
+            db_type.lower() in (
+                'tinyblob', 'blob', 'mediumblob', 'longblob', 'tinytext',
```

Comment:
Perhaps the tuple should be a module constant somewhere so it can be reused in `validation.py`.